src/service/service.ts

```diff
@@ -922,6 +922,10 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
           request: createFindNodeMessage([0]),
         });
       }
+      // If sendPing was initiated by the user and callback function set, call it and resolve the promise
+      if (activeRequest.callback) {
+          activeRequest.callback(null, message)
+      }
```

Comment:
I think this needs to be outside the `if (enr) {}` block.
Preference to putting this at the end of the function if possible.

Also, might be worth copying out only the necessary info in the returned message, eg: `activeRequest.calback(null, { enrSeq: message.enrSeq, addr: message.enrAddr });`